Projects/surface_photometry.ipynb

```diff
@@ -0,0 +1,1332 @@
+    "    mask[new_yind[in_im_y & in_im_x], new_xind[in_im_y & in_im_x]] = np.nan"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Checking that our resulting mask makes sense\n",
+    "fig, ax = plt.subplots(figsize=(21, 7), nrows=1, ncols=3)\n",
+    "ax[0].imshow(rgb_alpha, origin='bottom')\n",
+    "ax[0].set_title('RGBA IMAGE', fontsize=25)\n",
+    "ax[0].set_xlabel('X [pixels]', fontsize=20)\n",
+    "ax[0].set_ylabel('Y [pixels]', fontsize=20)\n",
+    "\n",
+    "# Copy the RGB image \n",
+    "rgb_alpha_copy = copy(rgb_alpha)\n",
+    "\n",
+    "for k in range(0, 4):\n",
+    "    rgb_alpha_copy[:,:,k] = rgb_alpha_copy[:,:,k] + mask \n",
+    "\n",
+    "ax[1].imshow(rgb_alpha_copy,  origin='bottom')\n",
+    "ax[1].set_title('MASK', fontsize=25)\n",
+    "ax[1].set_xlabel('X [pixels]', fontsize=20)\n",
+    "\n",
+    "for k in range(0, 4):\n",
+    "    rgb_alpha_copy[:,:,k] = rgb_alpha_copy[:,:,k] + sky_mask\n",
+    "\n",
+    "ax[2].imshow(rgb_alpha_copy,  origin='bottom')\n",
+    "ax[2].set_title('SKY MASK', fontsize=25)\n",
+    "ax[2].set_xlabel('X [pixels]', fontsize=20)\n",
+    "\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This shows that we can pick out pixels that belong to everything except the central galaxy and whatever surrounding galaxies are part of the same footprint (right panel) or that we can include the target galaxy mask as well (for background sky estimation, shown in the next section)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### **Deblending**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Of course, for a more correct mask we also want to include masks around sources overlapping the galaxy.  Because the current implementation of the deblender in the Stack is out of date, we're going to use the following package by Remy Joseph:\n",
+    "\n",
+    "https://github.com/herjy/scarlet_extensions\n",
+    "\n",
+    "To run this notebook from this point onward, you will need to do a git clone on this package in your working directory, then move the scarlet_extensions folder out from the newly created repository.  Also, this package requires SEP, a SExtractor protocol for Python.  This can be installed quite easily: `pip install sep` (however, make sure you do it in an LSST environment)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Scarlet deblending requires specifically a multiband exposure object to work with. We already created this for the `createRGB()` function, using the first three filters only."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images = coadds.image.array\n",
+    "seg_catalog, bg_rms = det.makeCatalog(images)"
```

Comment:
With newer versions of `scarlet_extensions` (after July 2020), makeCatalog expects a list of `Data` objects
```
images = coadds.image.array
datas = [det.Data(coadds.image.array,None,None,channels=bands[:3])]
seg_catalog, bg_rms = det.makeCatalog(datas)
```

Reply:
Looks like there is a recent tag (0.0.1) of scarlet_extensions

Reply:
So I think it would be 
```
git clone git@github.com:fred3m/scarlet_extensions.git
cd scarlet_extensions/
git checkout 0.0.1
cd ..
```